### ISSUE TITLE
Get error:The command '/bin/sh -c adduser studio' returned a non-zero code: 127

### ISSUE THREAD

**Issue by dingxiaoli** [felipecrs/docker-images#202]
add one docker file as follows:
--------------------------------------
FROM ghcr.io/felipecrs/jenkins-agent-dind:sha-f32dea2

RUN mkdir -p /home/jenkins/automation /home/jenkins/work && \
    chown jenkins:jenkins /home/jenkins/automation /home/jenkins/work

USER root

RUN apt-get update && apt-get install -y \
    unzip \
    wget \
    passwd \
    && rm -rf /var/lib/apt/lists/*

RUN apt-get update && apt-get install -y openjdk-17-jdk && rm -rf /var/lib/apt/lists/*

ENV JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64
ENV PATH="${JAVA_HOME}/bin:${PATH}"

RUN wget https://downloads.apache.org/maven/maven-3/3.9.6/binaries/apache-maven-3.9.6-bin.tar.gz -P /tmp && \
    tar -zxf /tmp/apache-maven-3.9.6-bin.tar.gz -C /usr/local && \
    mv /usr/local/apache-maven-3.9.6 /usr/local/maven-3.9.6

ENV MAVEN_HOME=/usr/local/maven-3.9.6 \
    MAVEN_CONFIG=$USER_HOME_DIR/.m2 \
    PATH="/usr/local/maven-3.9.6/bin:$PATH"

RUN wget https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip -P /tmp && \
    unzip /tmp/awscli-exe-linux-x86_64.zip -d /tmp && \
    /tmp/aws/install && \
    rm -rf /tmp/aws /tmp/awscli-exe-linux-x86_64.zip
	

RUN apt-get update && apt-get install -y \
    adduser \
    && rm -rf /var/lib/apt/lists/*

RUN adduser --disabled-password --gecos "" studio

USER jenkins

HEALTHCHECK --interval=30s --timeout=10s --start-period=30s --retries=3 CMD docker info > /dev/null 2>&1 || exit 1

--------------------------------------

then use this image on jenkins file:

 containers:
  - name: jnlp
    image: artifactory.datapwn.com/tlnd-docker-dev/talend/esb/appint/ciauto-dind-jdk17:latest
    
but get one error when run on pipeline jenkins:

`[INFO] k8s: Digest: sha256:f4ab40c615ec2bd6a09992f6ceebeca796752e58ea778ccc6439087a728ba3b3
[INFO] k8s: Status: Downloaded newer image for eclipse-temurin:11
[INFO] k8s: Pulled eclipse-temurin:11 in 3 seconds 
[INFO] k8s: [citest/rest/demorest_ms:latest]: Created docker-build.tar in 175 milliseconds
[ERROR] k8s: Failed to execute the build [Error while trying to build the image: Unable to build image [citest/rest/demorest_ms] : "The command '/bin/sh -c adduser studio' returned a non-zero code: 127" ]`

`[ERROR] Failed to execute goal org.eclipse.jkube:kubernetes-maven-plugin:1.12.0:build (start) on project VMSpringDemoMS: Failed to execute the build: Error while trying to build the image: Unable to build image [citest/vmspringdemoms] : "The command '/bin/sh -c adduser studio' returned a non-zero code: 127" -> [Help 1]`

**Comment by felipecrs** [comment]
I think ubuntu base image has no "adduser" command. Usually it's for alpine. You need to adjust your dockerfile with "useradd" instead.

**Comment by dingxiaoli** [comment]
![image](https://github.com/user-attachments/assets/68aec755-8c4b-4552-924d-bfb0b2e996ed)
related to this upgrade?

**Comment by felipecrs** [comment]
No, I meant to say `ghcr.io/felipecrs/jenkins-agent-dind` is based on Ubuntu, and on Ubuntu, you should use `useradd` command to create a new user rather than `adduser`:

```console
$ docker run --rm -it ubuntu adduser
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "adduser": executable file not found in $PATH: unknown.
```

**Comment by dingxiaoli** [comment]
Now clear the reason in my environment.
For the eclipse-temurin:11 Docker image, earlier versions may have included the shadow package, but in the latest version, this package is no longer included. 
thanks for above reply and help.

**Comment by felipecrs** [comment]
But `ghcr.io/felipecrs/jenkins-agent-dind` is not built from `eclipse-temurin:11`, it's built from `ubuntu`. And you are building your image from `ghcr.io/felipecrs/jenkins-agent-dind`, so I didn't understand why `eclipse-temurin:11` would matter.

**Comment by dingxiaoli** [comment]
You are right.
for the above docker file,it used to run Jenkins Agent.
but I have second docker file that is used to build and run a specific Java application container(I  did not notice it when I filed this issue,so I did not post the second docker file here),for the second docker file,it based on eclipse-temurin:11.Now everything is ok.
Sorry for filing this issue because my misunderstanding.
Thanks for above reply.
